### ISSUE TITLE
Galicaster preview freezes after multiples recordings

### ISSUE THREAD

**Issue by silviabf** [teltek/Galicaster#457]
Steps:

* Launch galicaster
* Schedule 200 recordings with the script, separated by 1 minutes: python docs/scripts/scheduler.py -o True -s https://server -u matterhorn_system_account -p CHANGE_ME -l 5 -d 1 -r 100
Problem: After a long number of recordings (130 or 150 in my tests) galicaster preview freezes when a recording stops, but it continues recording.
There is not info in the console neither in the logs.
Note: The number of recording when galicaster preview freezes depends of the used profile. It freezes earlier if I use a webcam profile that a blackmagic profile.

**Comment by ppettit** [comment]
i've been doing some testing to try and get to the bottom of this (it happens very often for me - normally after round 3-4 recordings - such that i am not sure we can use galicaster 2.0 in production).

trying to trace what is happening i noticed that the last thing that happens before the freeze appears to be recreating the display areas, which involves the `RecorderService` calling the `create_drawing_areas()` method on the `RecorderClassUI` so i thought maybe it is another threading issue (though not sure if they are actually in the same thread anyway?). unfortunately commenting out the drawing area stuff in `__prepare()` does not help, although the video preview (which is now displayed in a new window) does not freeze when the UI does.

it still looks like some kind of deadlock in the UI unrelated to the pipeline to me, but can't work out what.

is there any other progress or theories on this? do you have ny pointers on where else to look?

**Comment by Alfro** [comment]
@ppettit In our tests so far, we have noticed it does not happen when the profile does not have an audio track.

We believe it may be related with the "level" element of the audio pipelines. Lowering the interval property of this element seems to increase the likelihood of this happening, and removing it seems to stop it completely (with the unfortunate consequence of not having a VUmeter operative).

**Comment by ppettit** [comment]
thanks @Alfro, i'll give that a go to see if i can reproduce that.

**Comment by Alfro** [comment]
After doing this change b8c37e5 we haven't been able to reproduce this error again.

However, now the audio lost issue happens #387.

**Comment by ppettit** [comment]
i can confirm that the level element appear to have no impact on this. neither removing the element completely from the bin, nor increasing the number of level messages by 1000x neither seems to affect how quickly the freeze happens.

i'll try b8c37e5 now and see if that helps for me.

**Comment by Alfro** [comment]
@ppettit Do you mean the freeze consistently happens to you after 2-3 recordings even with no level element?
Could you attach your conf.ini/profile?

**Comment by ppettit** [comment]
b8c37e5 seems to have done the trick. i have not experienced a freeze since applying that, so now i always hit #387

**Comment by ppettit** [comment]
@Alfro no, nothing consistent. it appeared to be last week but i guess it was just random. this week i was sometimes doing 100+ recordings with no freeze (bot with and without the level element)
